app/javascript/controllers/logo_removal_controller.js

```diff
@@ -0,0 +1,14 @@
+import { Controller } from "@hotwired/stimulus"
+
+export default class extends Controller {
+  static targets = ["form"]
+
+  remove() {
+    const hidden = document.createElement('input')
+    hidden.type = 'hidden'
+    hidden.name = 'challenge[remove_logo]'
+    hidden.value = '1'
+    this.formTarget.appendChild(hidden)
+    this.formTarget.requestSubmit()
```

Comment:
This has a side effect - the whole form is submitted, so even unwanted changes made in the form are saved. 

Here I see 2 solutions:
  1. simple: a checkbox on the webpage (it can look like a toggle button) and a form submit made only when the "Save" button is pressed. This solution also eliminates a dedicated stimulus controller.
  2. Dedicated resource for uploading and deleting challenge icon - not connected with challenge edit.